lib/utils/makeValidator.js

```diff
@@ -18,6 +21,72 @@ const makeLinks = (error, makerFunc) => {
   return [];
 };
 
+const removeFirstAndLastChar = s => s.slice(1, -1);
+// always return a string
+const makePath = (path, newSegment) =>
+  (path ? [path, newSegment].join('.') : newSegment) || '';
+
+const processBaseError = (err, path) => {
+  const completePath = makePath(path, err.property).replace(
+    /\.instance\./g,
+    '.'
+  );
+
+  err.property = completePath;
+  return err;
+};
+
+const cleanError = (validationError, path, validator) => {
+  if (ambiguousTypes.includes(validationError.name)) {
+    // TODO: make this smarter?
+    // if it's triggerSchema, check for "type"
+    // if it's request/function, check for "source" or "require". if neither are present, validate against request
```

Comment:
(Non-blocker) For this TODO, do you think if it's possible to have a generic solution instead of relying on the prior knowledge we have on the app schema? I was thinking we can measure the "relevance" and find out the best match, similar to what python-jsonschema [does](https://github.com/Julian/jsonschema/blob/21838cd7727bd7c1d9a309df51cd32ebb0c78cdb/jsonschema/exceptions.py#L296).

Reply:
that... is very cool. It's a little weird because it looks like they just de-prioritize the `anyOf` errors. In that link, `STRONG_MATCHES` is an empty set, so i'm a little confused at how much value there is there. 

And yeah, i was going back and forth here. On the one hand, if we _only_ fish on polling/hook operation schema, we're already seeing a big improvement. doing more is even better, but i'd happily ship with just that.

Reply:
One idea I had was to recurse down each option (in this case `pollingOperation` and `hookOperation`) and then surface whichever path had fewer errors. Basically, the way we do it now, we could send someone down the wrong path if they do have a very malformed string request or something. Very possible I'm overthinking it though. 

My other thought is fix polling vs hook explicitly (since that's the most confusing one - it's usually an issue with an input field or something and it doesn't actually matter which of the two we validate against) and then putting a line in for all other cases of `anyOf` that says "Please file an issue if you see this so we can better cover this case". And then we can build out this section as needed.

what do you think?